File: python/verificador.py
import os

# Recebe o nome do arquivo via terminal
filename = input("Digite o nome do arquivo: ")
full_path = os.path.join(os.path.dirname(__file__), filename)

# Verifica se o arquivo existe
if not os.path.exists(full_path):
    print(f"Arquivo '{filename}' não encontrado no diretório atual.")
    exit()

try:
    with open(full_path, 'r') as file:
        soma_procedimentos_bpac = 0
        soma_procedimentos_bpai = 0

        # Ignora a primeira linha (cabeçalho)
        next(file)

        for line in file:
            header_line = line[0:2]

            # BPA-C
            if header_line == '02':
                number_procedimento_bpac = int(line[26:36])
                quantity_bpac = int(line[39:45])
                soma_procedimentos_bpac += number_procedimento_bpac + quantity_bpac

            # BPA-I
            if header_line == '03':
                number_procedimento_bpai = int(line[49:59])
                quantity_bpai = int(line[88:94])
                soma_procedimentos_bpai += number_procedimento_bpai + quantity_bpai

except IOError:
    print(f"Erro: o arquivo '{filename}' existe, mas não pôde ser aberto (verifique permissões).")
    exit()

# Cálculos finais
soma_total = soma_procedimentos_bpac + soma_procedimentos_bpai
resto_da_divisao = soma_total % 1111
campo_de_controle = resto_da_divisao + 1111

# Exibição dos resultados
print(f"A soma dos procedimentos BPA-C é: {soma_procedimentos_bpac}")
print(f"A soma dos procedimentos BPA-I é: {soma_procedimentos_bpai}")
print(f"A soma TOTAL dos procedimentos é: {soma_total}")
print(f"O resto da divisão do valor TOTAL dos procedimentos é: {resto_da_divisao}")
print(f"O campo de controle deve ser: {campo_de_controle}")

File: php/verificador.php
<?php
// Recebe a linha final para a primeira soma a partir do terminal
$filename = readline("Digite o nome do arquivo: ");
$fullPath = __DIR__ . '/' . $filename;

if (!file_exists($fullPath)) {
    echo "Arquivo '$filename' não encontrado no diretório atual.\n";
    exit;
}

$handle = fopen($fullPath, 'r');
if (!$handle) {
    echo "Erro: o arquivo '$filename' existe, mas não pôde ser aberto (verifique permissões).\n";
    exit;
}

$somaProcedimentosBpaC = 0;
$somaProcedimentosBpaI = 0;
// Ignora a primeira linha (cabeçalho)
fgets($handle);

while (($line = fgets($handle)) !== false) {
    // Pega os dois primeiros caracteres da linha
    $headerLine = substr($line, 0, 2);

    // BPA-C
    if ($headerLine === '02') {
        // Extrai o número da posição 27 a 36
        $numberProcedimentoBpaC = (int) substr($line, 26, 10);

        // Extrai o número da posição 40 a 45
        $quantityBpaC = (int) substr($line, 39, 6);

        // Soma ambos os números
        $somaProcedimentosBpaC += $numberProcedimentoBpaC + $quantityBpaC;
    }

    // BPA-I
    if ($headerLine === '03') {
        // Extrai o número da posição 50 a 59
        $numberProcedimentoBpaI = (int) substr($line, 49, 10);

        // Extrai o número da posição 89 a 94
        $quantityBpaI = (int) substr($line, 88, 6);

        // Soma ambos os números
        $somaProcedimentosBpaI += $numberProcedimentoBpaI + $quantityBpaI;
    }
}

fclose($handle);

$somaTotal = $somaProcedimentosBpaC + $somaProcedimentosBpaI;
$restoDaDivisao = $somaTotal % 1111;
$campoDeControle = $restoDaDivisao + 1111;

echo "A soma dos procedimentos BPA-C é: $somaProcedimentosBpaC\n";
echo "A soma dos procedimentos BPA-I é: $somaProcedimentosBpaI\n";
echo "A soma TOTAL dos procedimentos é: $somaTotal\n";
echo "O resto da divisão do valor TOTAL dos procedimentos é: $restoDaDivisao\n";
echo "O campo de controle deve ser: $campoDeControle\n";
